### PR TITLE
fix: RTE + Form + JSONForm reskinning

### LINKED ISSUE #10977: [Enhancement]: Form widget reskinning
https://www.notion.so/appsmith/Form-42fd8dcc1ecf43159b13393a95ddedf2

### LINKED ISSUE #10986: [Enhancement]: Rich text editor reskinning
https://www.notion.so/appsmith/Rich-text-editor-2edaa00c265f458c84cb244b65511902

### LINKED ISSUE #15171: JSON form reskinning
https://www.notion.so/appsmith/JSON-form-57ec14b1826c49498dc9203a8b5fdf46

### LINKED ISSUE #16879: [Bug]: Checkbox group widget - Red color for required remains even on disabling widget, and changes to grey on hover
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When Required is enabled and if all boxes are de-selected, and user disables the checkbox group widget, the red borders on the checkboxes remain and on hovering over the checkboxes, the colors change to grey from red. 

Expected behavior - The colors of the borders should also get greyed out as the widget is disabled. 

### Steps To Reproduce

1. Drag and drop a checkbox group widget
2. Set Required field to ON
3. Bind the Disabled toggle to - {{Checkbox2.isChecked}} and set Checkbox2 default state to off. 
4. Now uncheck all the checkboxes. We see the red borders on all the checkboxes.
5. Disable the widget. The red borders remain and change to grey on hover, even though the widget is disabled.
Chkbox_RequiredBug_Tobelogged

### Public Sample App

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes UI inconsistencies in RTE, Form and JSON Form reskinning. Also fixes one small color bug for checkbox in disabled state.

Fixes #15171
Fixes #10977
Fixes #10986
Fixes #16879

## Type of change

- Bug fix

## How Has This Been Tested?
- Manually tested

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by chandannkumar** [comment]
Tested this for:
- UI inconsistencies in RTE, Form and JSON Form reskinning
- Fixes one small color bug for checkbox in disabled state

**Review by dhruvikn — CHANGES_REQUESTED** [review]

File: app/client/src/widgets/FormWidget/index.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ export const CONFIG = {
   defaults: {
     rows: 40,
     columns: 24,
+    borderColor: Colors.GREY_5,
```
[the remark this answers is not in the document]
Comment (jsartisan) [reply app/client/src/widgets/FormWidget/index.ts:16]: We can't use css variables here :/ 
This config will require actual values as these will be used in property pane.

[the remark this answers is not in the document]
Comment (dhruvikn) [reply app/client/src/widgets/FormWidget/index.ts:16]: Ah my bad, looks good then!

**Review by dhruvikn — DISMISSED** [review]

**Review by dhruvikn — APPROVED** [review]

**Pull request merged**